From 1666342acf366456dc11ae6d208914745e978452 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Sun, 2 Nov 2025 12:49:27 +0200
Subject: [PATCH 1/1] device/marvell: add dummy platform.json avoiding WARNINGs

Marvell device platform/marvell-prestera uses schema with no
platform.json file.
Although this schema is working properly on the TRIXIE,
the systemd-sonic-generator is looking for this file and
gives the WARNING
  Failed to read /usr/share/sonic/device/<board>/platform.json

To avoid the warning add a dummy platform.json for each board.

Signed-off-by: Yan Markman <ymarkman@marvell.com>
---
 .../marvell/arm64-marvell_db98cx8580_32cd-r0/platform.json   | 5 +++++
 device/marvell/arm64-marvell_rd98DX35xx-r0/platform.json     | 5 +++++
 .../marvell/arm64-marvell_rd98DX35xx_cn9131-r0/platform.json | 5 +++++
 .../marvell/arm64-marvell_rd98DX45xx_cn9131-r0/platform.json | 5 +++++
 4 files changed, 20 insertions(+)
 create mode 100644 device/marvell/arm64-marvell_db98cx8580_32cd-r0/platform.json
 create mode 100644 device/marvell/arm64-marvell_rd98DX35xx-r0/platform.json
 create mode 100644 device/marvell/arm64-marvell_rd98DX35xx_cn9131-r0/platform.json
 create mode 100644 device/marvell/arm64-marvell_rd98DX45xx_cn9131-r0/platform.json

diff --git a/device/marvell/arm64-marvell_db98cx8580_32cd-r0/platform.json b/device/marvell/arm64-marvell_db98cx8580_32cd-r0/platform.json
new file mode 100644
index 000000000..e8c05f14b
--- /dev/null
+++ b/device/marvell/arm64-marvell_db98cx8580_32cd-r0/platform.json
@@ -0,0 +1,5 @@
+{
+    "platform": "arm64-marvell_db98cx8580_32cd-r0",
+    "hwsku": "default"
+}
+
diff --git a/device/marvell/arm64-marvell_rd98DX35xx-r0/platform.json b/device/marvell/arm64-marvell_rd98DX35xx-r0/platform.json
new file mode 100644
index 000000000..7ee41c5d8
--- /dev/null
+++ b/device/marvell/arm64-marvell_rd98DX35xx-r0/platform.json
@@ -0,0 +1,5 @@
+{
+    "platform": "arm64-marvell_rd98DX35xx-r0",
+    "hwsku": "default"
+}
+
diff --git a/device/marvell/arm64-marvell_rd98DX35xx_cn9131-r0/platform.json b/device/marvell/arm64-marvell_rd98DX35xx_cn9131-r0/platform.json
new file mode 100644
index 000000000..d4a4728dc
--- /dev/null
+++ b/device/marvell/arm64-marvell_rd98DX35xx_cn9131-r0/platform.json
@@ -0,0 +1,5 @@
+{
+    "platform": "arm64-marvell_rd98DX35xx_cn9131-r0",
+    "hwsku": "default"
+}
+
diff --git a/device/marvell/arm64-marvell_rd98DX45xx_cn9131-r0/platform.json b/device/marvell/arm64-marvell_rd98DX45xx_cn9131-r0/platform.json
new file mode 100644
index 000000000..1a9a02e54
--- /dev/null
+++ b/device/marvell/arm64-marvell_rd98DX45xx_cn9131-r0/platform.json
@@ -0,0 +1,5 @@
+{
+    "platform": "arm64-marvell_rd98DX45xx_cn9131-r0",
+    "hwsku": "default"
+}
+
-- 
2.34.1

